exporter/opentelemetry-exporter-otlp-proto-http/tests/metrics/test_otlp_metrics_exporter.py

```diff
@@ -0,0 +1,185 @@
+            OTEL_EXPORTER_OTLP_METRICS_ENDPOINT: "https://metrics.endpoint.env",
+            OTEL_EXPORTER_OTLP_HEADERS: OS_ENV_HEADERS,
+            OTEL_EXPORTER_OTLP_TIMEOUT: OS_ENV_TIMEOUT,
+        },
+    )
+    def test_exporter_constructor_take_priority(self):
+        exporter = OTLPMetricExporter(
+            endpoint="example.com/1234",
+            certificate_file="path/to/service.crt",
+            headers={"testHeader1": "value1", "testHeader2": "value2"},
+            timeout=20,
+            compression=Compression.NoCompression,
+            session=requests.Session(),
+        )
+
+        self.assertEqual(exporter._endpoint, "example.com/1234")
+        self.assertEqual(exporter._certificate_file, "path/to/service.crt")
+        self.assertEqual(exporter._timeout, 20)
+        self.assertIs(exporter._compression, Compression.NoCompression)
+        self.assertEqual(
+            exporter._headers,
+            {"testHeader1": "value1", "testHeader2": "value2"},
+        )
+        self.assertIsInstance(exporter._session, requests.Session)
+
+    @patch.dict(
+        "os.environ",
+        {
+            OTEL_EXPORTER_OTLP_CERTIFICATE: OS_ENV_CERTIFICATE,
+            OTEL_EXPORTER_OTLP_COMPRESSION: Compression.Gzip.value,
+            OTEL_EXPORTER_OTLP_HEADERS: OS_ENV_HEADERS,
+            OTEL_EXPORTER_OTLP_TIMEOUT: OS_ENV_TIMEOUT,
+        },
+    )
+    def test_exporter_env(self):
+
+        exporter = OTLPMetricExporter()
+
+        self.assertEqual(exporter._certificate_file, OS_ENV_CERTIFICATE)
+        self.assertEqual(exporter._timeout, int(OS_ENV_TIMEOUT))
+        self.assertIs(exporter._compression, Compression.Gzip)
+        self.assertEqual(
+            exporter._headers, {"envheader1": "val1", "envheader2": "val2"}
+        )
+
+    @patch.dict(
+        "os.environ",
+        {OTEL_EXPORTER_OTLP_ENDPOINT: OS_ENV_ENDPOINT},
+    )
+    def test_exporter_env_endpoint_without_slash(self):
+
+        exporter = OTLPMetricExporter()
+
+        self.assertEqual(
+            exporter._endpoint,
+            OS_ENV_ENDPOINT + f"/{DEFAULT_METRICS_EXPORT_PATH}",
+        )
+
+    @patch.dict(
+        "os.environ",
+        {OTEL_EXPORTER_OTLP_ENDPOINT: OS_ENV_ENDPOINT + "/"},
+    )
+    def test_exporter_env_endpoint_with_slash(self):
+
+        exporter = OTLPMetricExporter()
+
+        self.assertEqual(
+            exporter._endpoint,
+            OS_ENV_ENDPOINT + f"/{DEFAULT_METRICS_EXPORT_PATH}",
+        )
+
+    @patch.dict(
+        "os.environ",
+        {
+            OTEL_EXPORTER_OTLP_HEADERS: "envHeader1=val1,envHeader2=val2,missingValue"
+        },
+    )
+    def test_headers_parse_from_env(self):
+
+        with self.assertLogs(level="WARNING") as cm:
+            _ = OTLPMetricExporter()
+
+            self.assertEqual(
+                cm.records[0].message,
+                "Header doesn't match the format: missingValue.",
+            )
```

Comment:
Need some test which ensures the serialization done correctly and without any errors.

Reply:
created serialization and http response tests